app/services/builders/chartkick/helpers/success_rate.rb

```diff
@@ -0,0 +1,63 @@
+module Builders
+  module Chartkick
+    module Helpers
+      class SuccessRate < BaseService
+        attr_reader :department_name, :metric_name, :intervals
+
+        RateDetails = Struct.new(:rate, :successful, :total, :metric_detail)
+        MetricSetting = Struct.new(:name, :value)
+
+        def initialize(department_name, metric_name, intervals)
+          @department_name = department_name
+          @metric_name = metric_name.to_s
+          @intervals = intervals
+        end
+
+        def call
+          return if total.zero?
+
+          RateDetails.new(rate, successful, total, metric_detail)
+        end
+
+        private
+
+        def metric_detail
+          MetricSetting.new(metric_key, metric_setting)
+        end
+
+        def total
+          @total ||= intervals.map { |interval| interval[1] }
+                              .reduce(0) { |sum, num| sum + num }
```

Comment:
Maybe you can use `sum` instead of `reduce`? https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-sum